### ISSUE TITLE
Closing application takes substantial time

### ISSUE THREAD

**Issue by AnshKetchum** [ClimenteA/flaskwebgui#44]
I created an application using Flask and used flaskwebgui to convert it into a desktop application.

When I directly run my ``app.py`` file, it takes around a 20-30 seconds after the window is closed for the program to finally terminate. 

When I run an executable created by Pyinstaller using the Terminal, even after I close my window, the thread never stops / the terminal never closes. 

What do I need to change in the code such that the thread closes (executes an ``on_exit`` method that shelves 10 variables, then terminates) immediately after I close (press the 'X' on my GUI)? Are there any sleeps / thread pausing that I can delete?

**Comment by ClimenteA** [comment]
Figuring out which chrome window belongs to flaskwebgui was difficult to implement, because I can't get the PID from the chrome window opened.

As a workaround you see in docs that we put a js code which asks the server if it's open each 5 seconds. That is request goes to server and the server writes to a file the timestamp, if more than 5 seconds passed since last js request it means the chrome window was closed and then the server closes also (see source code).

I had some issues with pyinstaller in the past not only with flaskwebgui but with other stuff also that's why I created pyvan.

See if packaging it with pyvan solves this issue.

**Comment by AnshKetchum** [comment]
Hi ClimenteA, thank you for taking time to help me. 

I tried ``pip install pyvan`` from a Python3.7 virtual environment, but it didn't work. I got this error: 

```
ERROR: Could not find a version that satisfies the requirement pywin32 (from gen-exe->pyvan) (from versions: none)
ERROR: No matching distribution found for pywin32 (from gen-exe->pyvan)
```
(Maybe it is because I am using MacOS??)

Also, I am using a flask application that relies on a virtual env, has a static and templates folder. They are in the same directory as my ``app.py``. How would I tell pyvan to encorporate these directories assuming the installation error is fixed?

**Comment by ClimenteA** [comment]
Unfortunately pyvan works only for windows..

**Comment by AnshKetchum** [comment]
Hi ClimenteA, thank you for your reply.

Is there any way I can reduce the time after closing by changing code?

**Comment by AnshKetchum** [comment]
I took a closer look at the code and noticed that you have 3 threads, but the thread for closing the window feels unecessary to me. Is there any way you could replace the third thread with some "on_close" function that executes when the thread that open's the browser closes? 

That way you don't need to keep on checking for connection.

**Comment by ClimenteA** [comment]
Removing one thread will not fix the issue. Your issue is related to this https://github.com/ClimenteA/flaskwebgui/issues/42

Not sure why it raises a BrokenPipe error, I will take a look the following days

**Comment by AnshKetchum** [comment]
Hi ClimenteA,

Thank you for helping me. I wrote the comment as suggestions to improve efficiency (there is no issue / ``BrokenPipeError``, the code is just super duper slow) of the code and prevent the need to have to unnecessarily create a third thread. 

This suggestion was given in mind to improve the efficiency of your program since having an ``on-close `` method instead of starting redundant threads would reduce the number of times that set of code is executed (and also removes  the need to check if the browser still exists, sleep for 2 seconds, repeat,  etc, which would improve the speed of program after closing).

**Comment by ClimenteA** [comment]
Maybe instead of polling a websocket connection would be better. That way you don't wait 10 seconds for the server to close.

**Comment by AnshKetchum** [comment]
Hi ClimenteA,

I am not sure what you mean. Are you saying to create a ``SocketIO`` instance and add it to the ``FlaskUI`` object?

**Comment by ClimenteA** [comment]
Released a new version(0.1.15+), hopefully it fixes this issue.
